src/utils/include/utils/LCA.hpp

```diff
@@ -164,13 +167,18 @@ inline int LCA::getLCA( int u, int v )
     return m_euler[queryRMQ( m_firstAppearance[u], m_firstAppearance[v] )];
 }
 
-inline std::string LCA::getLCA( std::vector< std::string >& taxIds )
+// TODO: this function should be const! For that, we cannot use operator[] for the maps. They
+// are not const and can potentially modify the maps. We should maybe call m_encode.at(...).
+inline std::string LCA::getLCA( const std::vector< std::string >& taxIds )
 {
-    int lca;
-    lca = getLCA( m_encode[taxIds[0]], m_encode[taxIds[1]] );
-    for ( unsigned int i = 2; i < taxIds.size(); i++ )
+    assert( taxIds.size() > 1 ); // I wonder if, aside from asserting this, if we should just branch off...
+
+    int lca = getLCA( m_encode[taxIds[0]], m_encode[taxIds[1]] );
+
+    for ( auto it = std::next( taxIds.begin(), 2 ); it != taxIds.end(); ++it )
     {
-        lca = getLCA( lca, m_encode[taxIds[i]] );
+        lca = getLCA( lca, m_encode[*it] );
     }
+
     return m_decode[lca];
```

Comment:
As we discussed, looks like `m_decode` could be a vector. I am just commenting here so we keep this idea floating around...